controller/controller.go

```diff
@@ -44,8 +46,11 @@ func IsJSONRequest(contentType string) bool {
 }
 
 func initExtensionUpdatesFromDynamoDB() {
-	sess, err := session.NewSession(&aws.Config{})
-
+	var awsConfig = aws.Config{}
+	if endpoint := os.Getenv("DYNAMODB_ENDPOINT"); endpoint != "" {
+		awsConfig.Endpoint = aws.String(endpoint)
+	}
+	sess, err := session.NewSession(&awsConfig)
```

Comment:
A couple of notes/nitpicks here:
1. Nit: no need for the `var` keyword. Also, can initialise and take the address in one step – this communicates the intent for cases where a pointer to a struct is being used.
2. The environment variable is hidden. I would suggest obtaining the value in `main` and passing it in here.

```suggestion
	awsConfig := &aws.Config{}
	if endpoint := os.Getenv("DYNAMODB_ENDPOINT"); endpoint != "" {
		awsConfig.Endpoint = aws.String(endpoint)
	}
	sess, err := session.NewSession(awsConfig)
```